addon/components/inputs/datetime.js

```diff
@@ -57,8 +69,47 @@ export default AbstractInput.extend({
     return moment(value).format(DATE_TIME_FORMAT)
   },
 
+  // == Lifecycle Hooks =======================================================
+
+  init () {
+    this._super(...arguments)
+
+    const value = this.get('value')
+    const timezone = this.get('timezone')
+    let currentDateTime = getMomentTimezone(value, timezone)
+
+    if (!this.get('defaultToCurrentDateTime')) {
+      currentDateTime = value
+    } else {
+      currentDateTime = getFormattedTime(currentDateTime, this.get('dateTimeFormat'), timezone)
```

Comment:
If I'm reading this correctly, it says if `defaultToCurrentDateTime` is true, then it always overrides the value to the current datetime. I don't think that's what we want. We only want to default it when there is no value.